### ISSUE TITLE
Bug or misleading documentation in the Getting Started guide about Structs

### ISSUE THREAD

**Issue by a115** [elixir-lang/elixir-lang.github.com#422]
In the getting started guide about Structs (http://elixir-lang.org/getting_started/15.html) it says the following:

    iex> Map.merge(%User{age: 27}, %User{name: "takashi"})
    %User{age: 27, name: "takashi"}

This leads a reader to believe that you can merge two structs with the Map.merge function, and you would get a new struct, where they keys of both would be combined. However, this seems to only work because the User struct is defined with a default age of 27. If, instead, you try this: 

    iex> Map.merge(%User{age: 28}, %User{name: "takashi"})
    %User{age: 27, name: "takashi"}

You get the wrong age (the default value of 27, instead of the 28 provided in the first argument to Map.merge).  

Not sure if this is a problem with the documentation or the language itself.

**Comment by alco** [comment]
It is a documentation issue. The example should probably be

```
Map.merge(%User{age: 28}, %{name: "takahashi"})
```
although using `Map.merge` is not safe in general, as it can potentially add keys that were not defined for the struct.

**Comment by whatyouhide** [comment]
The documentation of [`Dict.merge`](http://elixir-lang.org/docs/master/elixir/Dict.html#merge/3), which is what `Map.merge/2-3` implements, states that, in case of conflict, the keys from the dict passed as the second argument have higher precedence.

Using simple maps, you can easily see that:

```elixir
iex> map = %{a: 1, b: 2}
%{a: 1, b: 2}
iex> Map.merge map, %{a: "precedence"}
%{a: "precedence", b: 2}
```

Since `%User{name: "takashi"}` evaluates to `%User{age: 27, name: "takashi"}`, the `:age` of the second user takes precedence on the age of the first user. This is definitely not a language bug.

However, I agree that it's a documentation issue, like @alco said. I'll try to open a PR for this as soon as I can. :)

**Comment by whatyouhide** [comment]
I fixed this in #426, which has been merged. Maybe we can close this issue? :smiley:
